src/Lib.hs

```diff
@@ -231,6 +215,22 @@ getAttachmentsFromComment comments = do
     isAttachmentZip :: Attachment -> Bool
     isAttachmentZip attachment = "application/zip" == aContentType attachment
 
+-- | Get zendesk responses
+-- | Returns with maybe because it could return no response
+getZendeskResponses :: [Comment] -> [Attachment] -> TicketInfo -> App (Maybe ZendeskResponse)
+getZendeskResponses comments attachments ticketInfo
+    | not (null attachments) = inspectAttachments ticketInfo attachments
+    | not (null comments)    = Just <$> responseNoLogs ticketInfo
+    | otherwise              = return Nothing
+
+-- | Inspect only the latest attchment
+inspectAttachments :: TicketInfo -> [Attachment] -> App (Maybe ZendeskResponse)
+inspectAttachments ticketInfo attachments = do
+    -- Inspecting the last element since it'll be the latest attachment
+    let latestAttachment = last <$> nonEmpty attachments
```

Comment:
How about `inspectAttachment ticketInfo =<< safeHead . sort $ attachments` where the `Ord` instance should be `AttachmentId`? Or the expanded form of the same which is even better:
```
lastAttachment <- safeHead . sort $ attachments
inspectAttachment ticketInfo lastAttachment
```